src/dex/uniswap-v2/ring-v2.ts

```diff
@@ -0,0 +1,371 @@
+import { defaultAbiCoder, Interface } from '@ethersproject/abi';
+import { getAddress } from '@ethersproject/address';
+import { keccak256 } from '@ethersproject/keccak256';
+import {
+  DEST_TOKEN_PARASWAP_TRANSFERS,
+  Network,
+  SRC_TOKEN_PARASWAP_TRANSFERS,
+  SwapSide,
+} from '../../constants';
+import { IDexHelper } from '../../dex-helper';
+import { SpecialDex } from '../../executor/types';
+import { applyTransferFee } from '../../lib/token-transfer-fee';
+import {
+  Token,
+  TransferFeeParams,
+  ExchangePrices,
+  Address,
+  DexExchangeParam,
+  NumberAsString,
+  DexConfigMap,
+} from '../../types';
+import { getBigIntPow, getDexKeysWithNetwork, isETHAddress } from '../../utils';
+import { DexParams, UniswapData, UniswapV2Data } from './types';
+import { UniswapV2 } from './uniswap-v2';
+import ringV2ABI from '../../abi/ring-v2/ring-v2-pool.json';
+import ringV2factoryABI from '../../abi/ring-v2/ring-v2-factory.json';
+import RingV2ExchangeRouterABI from '../../abi/ring-v2/ring-v2-router.json';
+import ETHMainnetFewWrappedTokenJSON from '../../abi/ring-v2/few-wrapped-token.json';
+import { extractReturnAmountPosition } from '../../executor/utils';
+
+export enum RingV2Functions {
+  swapExactTokensForTokens = 'swapExactTokensForTokens',
+  swapExactETHForTokens = 'swapExactETHForTokens',
+  swapTokensForExactTokens = 'swapTokensForExactTokens',
+  swapTokensForExactETH = 'swapTokensForExactETH',
+  swapETHForExactTokens = 'swapETHForExactTokens',
+}
+
+const RingV2Config: DexConfigMap<DexParams> = {
+  RingV2: {
+    [Network.MAINNET]: {
+      subgraphURL: '2f5DMnspUwMx2n3229koTsFrAZRua3YV69pucVzWmQA9',
```

Comment:
Not sure if PoolTracker is going to work properly, as [subgraph](https://thegraph.com/explorer/subgraphs/2f5DMnspUwMx2n3229koTsFrAZRua3YV69pucVzWmQA9?view=Query&chain=arbitrum-one) returns pairs with fw tokens, so we won't receive original tokens anyway (e.g. DAI, USDC, WETH, etc)

So, I suggest removing subgraph support here to exclude PoolTracker on this dex, or we can also extend PoolTracker method to unwrap fw tokens to original ones (probably not worth it)